src/processors/default/createERC721NFTsFromTransfersProcessor.ts

```diff
@@ -31,43 +32,68 @@ const processorFunction = (contracts: NftFactory[]) =>
         throw 'buildNFTId not specified'
       }
 
+      const contractAddress = formatAddress(contract.id);
+      const fromAddress = formatAddress(item.args!.from);
+      const toAddress = formatAddress(item.args!.to);
       const tokenId = BigInt((item.args!.tokenId as BigNumber).toString());
-      const newMint = item.args!.from === ETHEREUM_NULL_ADDRESS;
+      const nftId = contractType.buildNFTId(contractAddress, tokenId);
+
       transfers.push({
-        id: `${CHAIN}/${item.blockNumber}/${item.logIndex}`,
-        contractAddress: formatAddress(contract.id),
-        from: item.args!.from,
-        to: item.args!.to,
+        id: ethereumTransferId(item.blockNumber, item.logIndex),
+        contractAddress: contractAddress,
+        from: fromAddress,
+        to: toAddress,
         tokenId,
         createdAtEthereumBlockNumber: '' + item.blockNumber,
-        nftId: contractType.buildNFTId(contract.id, tokenId), 
+        nftId: nftId,
         transactionHash: item.transactionHash
       });
-      if (!newMint) {
-        updates.push({
-          id: contractType.buildNFTId(contract.id, tokenId),
-          owner: item.args!.to
+
+      if (!newMint(fromAddress)) {
+        updatedNFTs.push({
+          id: nftId,
+          owner: toAddress
         })
+        updatedNftsCollectors.push({ nftId: nftId, collectorId: fromAddress, amount: -1 })
+        updatedNftsCollectors.push({ nftId: nftId, collectorId: toAddress, amount: 1 })
         return undefined;
       }
       newNFTs.push({
-        id: contractType.buildNFTId(contract.id, tokenId),
+        id: nftId,
         createdAtEthereumBlockNumber: '' + item.blockNumber,
-        contractAddress: formatAddress(contract.id),
+        contractAddress: contractAddress,
         tokenId,
         platformId: contract.platformId,
-        owner: item.args!.to,
+        owner: toAddress,
         approved: contract.autoApprove
       });
+      updatedNftsCollectors.push({ nftId: nftId, collectorId: toAddress, amount: 1 })
```

Comment:
Minted NFT -> +1 to receiver

Reply:
you're already doing this above for the !newMint scenario, so won't this happen twice in that scenario? should this line be in an else in the above conditional?

Reply:
Ah, I thought the `return undefined` on 59 was breaking us out of the loop?

I've moved it to an `else` statement for readability either way.

Reply:
oh yeah i missed that, sorry